projects/core/src/user/facade/order-return-request.service.ts

```diff
@@ -0,0 +1,99 @@
+import { Injectable } from '@angular/core';
+import { select, Store } from '@ngrx/store';
+import { Observable } from 'rxjs';
+import { map, take, tap } from 'rxjs/operators';
+import { AuthService } from '../../auth/facade/auth.service';
+import {
+  ReturnRequestList,
+  ReturnRequestEntryInputList,
+  ReturnRequest,
+} from '../../model/order.model';
+import { StateWithProcess } from '../../process/store/process-state';
+import { UserActions } from '../store/actions/index';
+import { UsersSelectors } from '../store/selectors/index';
+import { StateWithUser } from '../store/user-state';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class OrderReturnRequestService {
+  constructor(
+    protected store: Store<StateWithUser | StateWithProcess<void>>,
+    protected authService?: AuthService
+  ) {}
+
+  /**
+   * Create order return request
+   * @param orderCode an order code
+   * @param returnRequestInput order return request entry input
+   */
+  createOrderReturnRequest(
+    returnRequestInput: ReturnRequestEntryInputList
+  ): void {
+    this.authService
+      .getOccUserId()
+      .pipe(take(1))
+      .subscribe(userId =>
+        this.store.dispatch(
+          new UserActions.CreateOrderReturnRequest({
+            userId,
+            returnRequestInput,
+          })
+        )
+      )
+      .unsubscribe();
+  }
+
+  /**
+   * Return an order return request
+   */
+  getOrderReturnRequest(): Observable<ReturnRequest> {
+    return this.store.pipe(select(UsersSelectors.getOrderReturnRequest));
+  }
+
+  /**
+   * Gets order return request list
+   */
+  getOrderReturnRequestList(pageSize: number): Observable<ReturnRequestList> {
+    return this.store.pipe(
+      select(UsersSelectors.getOrderReturnRequestListState),
+      tap(returnListState => {
+        const attemptedLoad =
+          returnListState.loading ||
+          returnListState.success ||
+          returnListState.error;
+        if (!attemptedLoad) {
+          this.loadOrderReturnRequestList(pageSize);
+        }
```

Comment:
Shouldn't you filter to make sure there is a value?

Reply:
the initial value is `{returnReqeusts: [], pagination: {}, sorts:[]}`.